libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -147,13 +156,14 @@ void CodeTransform::freeUnusedVariables(bool _popUnusedSlotsAtStackTop)
 	if (!m_allowStackOpt)
 		return;
 
-	for (auto const& identifier: m_scope->identifiers)
-		if (holds_alternative<Scope::Variable>(identifier.second))
-		{
-			Scope::Variable const& var = std::get<Scope::Variable>(identifier.second);
-			if (m_variablesScheduledForDeletion.count(&var))
-				deleteVariable(var);
-		}
+	if (!m_returnStackHeight.has_value())
```

Comment:
The connection between the return stack height not being set and the fact that variables can be deleted is unclear to me. Maybe introduce a helper member function that returns a bool and has a descriptive name?

Reply:
I'm not sure what would be a more descriptive name - the idea is *exactly* that: we can delete variables outside the current scope *only* in the case before the return variables are allocated...

Reply:
Then what about a function called `allReturnVariablesAllocated`? This line only talks about something called a return stack height.